src/amber/router/pipe/session.cr

```diff
@@ -3,47 +3,78 @@ require "base64"
 require "yaml"
 require "openssl/hmac"
 
-module Amber::Pipe
-  # The session handler provides a cookie based session.  The handler will
-  # encode and decode the cookie and provide the hash in the context that can
-  # be used to maintain data across requests.
-  class Session < Base
-    property secret : String
-    property key : String
-    property store : Symbol
-
-    def initialize(
-      @key : String = "#{Server.settings.project_name}.session",
-      @secret : String = Server.settings.secret,
-      @store : Symbol = :cookie)
-    end
+module Amber
+  module Pipe
+    # The session handler provides a cookie based session.  The handler will
+    # encode and decode the cookie and provide the hash in the context that can
+    # be used to maintain data across requests.
+    class Session < Base
+      property secret : String
+      property key : String
+      property store : Symbol
```

Comment:
The different store will probably end up becoming a different handler since it will be parsing and saving sessions to redis, memcache, or some other sort of data storage.